campaignion_tracking/package/src/fragment.js

```diff
@@ -57,35 +63,40 @@ export function consumeLocationHashForPrefixes (prefixes = [], locationHash = ''
 }
 
 /**
- * Parse a part from the URL fragment.
- *
- * The optional "id" in the match item is the empty string if no id was given.
- *
- * TODO: optional multiple ids
- * TODO: whitespace trimming?
+ * Parse a single key-value part from the URL fragment.
  *
  * @param {String} part A part from the URL fragment.
  *
- * @returns {Object} A match item with the keys "prefix", "id", "codes", "origParts".
+ * @returns {Object} A match item with the keys "id", "codes", "origPart".
  */
 export function parsePart (part) {
   const _pos1 = part.indexOf('=')
   const value1 = part.substring(0, _pos1)
   const value2 = part.substring(_pos1 + 1)
-  let id = ''
-  let prefix = value1
-  // if there is a ':' we extract the id
-  const _pos2 = value1.indexOf(':')
-  if (_pos2 >= 0) {
-    prefix = value1.substring(0, _pos2)
-    id = value1.substring(_pos2 + 1)
-  }
   const codes = value2.split(',')
-
   return {
-    prefix: prefix,
-    id: id,
+    id: value1,
     codes: codes,
     origPart: part
   }
 }
+
+/**
+ * Parse a possible multiple parts from the URL fragment.
+ *
+ * Optionally tag all parts with a commong prefix.
+ *
+ * @param {String} part Parts from the URL fragment, delimited with `&`.
+ * @param {String} prefix A prefix to tag the parts with
+ *
+ * @returns {Array} A list of match objects
+ */
+export function parsePartsWithPrefix (partsString, prefix = '') {
+  const split = partsString.split('&')
+  const parts = split.map(part => {
+    const _p = parsePart(part)
```

Comment:
Does this handle URL-encoding? (I’m not sure where’s the best place to add the comment)

Reply:
Should be not an issue wrt `&`, they are not escaped:

```
> window.location.hash = '& '
"& "
> window.location.hash
"#&%20"
```

(for Fx and Chroumium)

Reply:
The parser still needs to handle those values properly and pass them decoded to the events.

How about `w:title=One: & two; & three,`?

Reply:
OK, thanks, got it...

Reply:
Should be fixed now.
- Assuming all values need to be encoded
- encoding only relevant string here: `title` (the other values are hardcoded chars or integers)